odpf/entropy/v1beta1/service.proto

```diff
@@ -0,0 +1,98 @@
+syntax = "proto3";
+
+package odpf.entropy.v1beta1;
+
+import "google/api/annotations.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/struct.proto";
+
+option go_package = "github.com/odpf/proton/entropy/v1beta1;entropyv1beta1";
+option java_multiple_files = true;
+option java_package = "io.odpf.proton.entropy";
+option java_outer_classname = "EntropyServiceProto";
+
+
+service ResourceService {
+  rpc ListResources(ListResourcesRequest) returns (ListResourcesResponse) {
+    option (google.api.http) = {
+      get: "/v1beta1/resources"
+    };
+  }
+
+  rpc GetResource(GetResourceRequest) returns (GetResourceResponse) {
+    option (google.api.http) = {
+      get: "/v1beta1/resources/{id}"
+    };
+  }
+
+  rpc CreateResource(CreateResourceRequest) returns (CreateResourceResponse) {
+    option (google.api.http) = {
+      post: "/v1beta1/resources"
+      body: "resource"
+    };
+  }
+
+  rpc UpdateResource(UpdateResourceRequest) returns (UpdateResourceResponse) {
+    option (google.api.http) = {
+      patch: "/v1beta1/resources/{resource.id}"
+      body: "resource"
+    };
+  }
+
+  rpc DeleteResource(DeleteResourceRequest) returns (DeleteResourceResponse) {
+    option (google.api.http) = {
+      delete: "/v1beta1/resources/{id}"
+    };
+  }
+}
+
+message Resource {
+  string id = 1;
+  string name = 2;
+  Module module = 3;
+  string parent = 4;
+  google.protobuf.Value configs = 5;
+}
+
+message Module {
+  string id = 1;
+}
+
+message ListResourcesRequest {
+  string parent = 1;
```

Comment:
doesn't resource here makes sense? Will allow us to search based on any key?

Reply:
As of now GRPC gateway doesn't support using a FieldMask and automatically updating it based on the query params for a get request.
There is an open issue for the same, once it is available we can move to that.